ui/app/components/ui/mascot/mascot.component.js

```diff
@@ -46,6 +69,32 @@ export default class Mascot extends Component {
 
   componentDidMount () {
     this.mascotContainer.current.appendChild(this.logo.container)
+    this.directionTargetMap = directionTargetGenerator(this.mascotContainer.current.getBoundingClientRect())
+
+    const { lookAtTarget, lookAtDirection } = this.props
+
+    if (lookAtTarget?.x && lookAtTarget?.y) {
+      this.logo.lookAtAndRender(lookAtTarget)
+    }
+    if (lookAtDirection) {
+      this.logo.lookAtAndRender(this.directionTargetMap[lookAtDirection])
+    }
+  }
+
+  componentDidUpdate (prevProps) {
+    const { lookAtTarget: prevTarget = {}, lookAtDirection: prevDirection = '', followMouse: prevFollowMouse } = prevProps
+    const { lookAtTarget = {}, followMouse, lookAtDirection } = this.props
+
+    if (lookAtDirection && prevDirection !== lookAtDirection) {
+      this.logo.lookAtAndRender(this.directionTargetMap[lookAtDirection])
+    }
+    if (lookAtTarget?.x !== prevTarget?.x || lookAtTarget?.y !== prevTarget?.y) {
+      this.logo.lookAtAndRender(lookAtTarget)
+    }
+    if (prevFollowMouse !== followMouse) {
+      this.unfollowMouse()
+      followMouse && this.refollowMouse()
+    }
```

Comment:
As: https://github.com/MetaMask/metamask-extension/pull/9166/files#r473383307

Reply:
You're right, it should be if/else. Fixed in 4ff2b1138